### DISCUSSION TITLE
Header Table with Merge Cells

### DISCUSSION THREAD

**Discussion by senzlord** [py-pdf/fpdf2#373] [Q&A]
I already make it, but is there any better way to make Header Table like Merge cells in sheets / excel?

Code I Use:
   ```python
    pdf = FPDF()
    pdf.add_page(orientation="landscape", format="A4")
    pdf.set_font('helvetica', size=12)
    line_height = pdf.font_size * 2
    col_width = pdf.epw / 10
    #Header
    pdf.set_font(style="B")
    pdf.cell(col_width, line_height*2, 'Event', border=2, align='C')
    pdf.cell(col_width*1.3, line_height*2, 'Agent', border=2, align='C')
    pdf.cell(col_width, line_height*2, 'Total Tickets', border=2, align='C')
    col = pdf.x - pdf.l_margin
    pdf.cell(col_width/1.5*3, line_height, 'Total Winning Tickets', border=2, align='C')
    pdf.cell(col_width, line_height*2, 'Total Bets', border=2, align='C')
    pdf.cell(col_width*1.5, line_height*2, 'Total Prizes', border=2, align='C')
    colfoot = pdf.x - pdf.l_margin
    pdf.cell(col_width*1.5, line_height*2, 'P/L', border=2, align='C')
    pdf.ln(line_height)
    pdf.cell(col)
    pdf.cell(col_width/1.5, line_height, 'Gold', border=2, align='C')
    pdf.cell(col_width/1.5, line_height, 'Silver', border=2, align='C')
    pdf.cell(col_width/1.5, line_height, 'Bronze', border=2, align='C')
    pdf.ln(line_height)
```

The one I already make:
![image](https://user-images.githubusercontent.com/22251174/160061577-e1aa3ad6-64ce-498b-a87f-b46d2e3c2199.png)

**Comment by Lucas-C** [comment]
Looks great to me :)

I'll just point you to other threads where you can find alternative approaches:
* https://pyfpdf.github.io/fpdf2/Tables.html#repeat-table-header-on-each-page
* https://github.com/PyFPDF/fpdf2/issues/210

**Comment by gmischler** [comment]
> I already make it, but is there any better way to make Header Table like Merge cells in sheets / excel?

Actually, this is pretty much exactly how I do it (plus the code up front to determine relative widths and heights that you left away above but probably have as well).
There is currently no more fully automated way in fpfd. IIrc, `write_html()` does colspan, but not rowspan, let alone a combination of the two.
